pkg/cli/parse_quotas.go

```diff
@@ -21,102 +21,237 @@ package cli
+		}
+		input.service = srvRes[0]
+		input.resource = srvRes[1]
 
-	// rxMatchedList: []string{"entire regex matched string", "quota value", "unit (empty, if no unit given)"}
-	rxMatchedList := rx.FindStringSubmatch(valCom[0])
-	intVal, err := strconv.ParseInt(rxMatchedList[1], 10, 64)
-	if err != nil {
-		return fmt.Errorf("could not parse quota value: '%s'", rxMatchedList[1])
-	}
-	tmp.Value = intVal
-
-	tmp.Unit = limes.UnitNone
-	if rxMatchedList[2] != "" {
-		switch rxMatchedList[2] {
-		case "B":
-			tmp.Unit = limes.UnitBytes
-		case "KiB":
-			tmp.Unit = limes.UnitKibibytes
-		case "MiB":
-			tmp.Unit = limes.UnitMebibytes
-		case "GiB":
-			tmp.Unit = limes.UnitGibibytes
-		case "TiB":
-			tmp.Unit = limes.UnitTebibytes
-		case "PiB":
-			tmp.Unit = limes.UnitPebibytes
-		case "EiB":
-			tmp.Unit = limes.UnitExbibytes
-		default:
-			return fmt.Errorf("acceptable units: ['B', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB'], got '%s'", rxMatchedList[2])
+		// separate quota value and comment (if one was given)
+		valCom := strings.SplitN(idfVal[1], ":", 2)
+		if len(valCom) > 1 {
+			if valCom[1] != "" {
+				input.comment = strings.TrimSpace(valCom[1])
+			}
+		}
+
+		// separate quota's value from its unit (if one was given)
+		rx := regexp.MustCompile(`^([0-9]*.[0-9]+)([A-Za-z]+)?$`)
+		match := rx.MatchString(valCom[0])
+		if !match {
+			return nil, fmt.Errorf("expected a quota value with optional unit in the format: 12.3Unit, got '%s'", valCom[0])
+		}
+
+		// rxMatchedList contains ["entire regex matched string", "quota value", "unit (empty, if no unit given)"]
+		rxMatchedList := rx.FindStringSubmatch(valCom[0])
+		input.value = rxMatchedList[1]
+
+		input.unit = limes.UnitNone
+		if rxMatchedList[2] != "" {
+			unitIsValid := false
+			for k := range quotaUnits {
+				if rxMatchedList[2] == string(k) {
+					input.unit = k
+					unitIsValid = true
+				}
+			}
+
+			if !unitIsValid {
+				return nil, fmt.Errorf("acceptable units: ['B', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB'], got '%s'", rxMatchedList[2])
+			}
+		}
+
+		if input.unit == limes.UnitNone {
+			if strings.Contains(input.value, ".") {
+				return nil, fmt.Errorf("counted values must be an integer, got '%s'", input.value)
+			}
+		}
+
+		_, exists := resUnits[input.service]
+		if !exists {
+			resUnits[input.service] = make(map[string]limes.Unit)
 		}
+
+		resUnits[input.service][input.resource] = input.unit
+
+		userInputs = append(userInputs, input)
 	}
 
-	(*q)[srv] = append((*q)[srv], tmp)
+	s.setBaseUnits(&resUnits)
+
+	for _, input := range userInputs {
+		var intQuotaVal int64
+
+		if strings.Contains(input.value, ".") {
+			diffInExp := quotaUnits[input.unit] - quotaUnits[resUnits[input.service][input.resource]]
+			if diffInExp < 0 {
+				return nil, fmt.Errorf("minimum accepted unit for '%s/%s' is '%v', got '%v'",
+					input.service, input.resource, resUnits[input.service][input.resource], input.unit)
+			}
```

Comment:
This paragraph could use an explanation comment, e.g. `fractional values are only possible when their unit is bigger than the base unit`.